orchestrator/relayer/src/fee_manager.rs

```diff
@@ -0,0 +1,219 @@
+                                                  pair = value);
+
+                        self.token_api_url_map.insert(key, request_url);
+                    }
+                }
+
+                return true;
+            }
+            RelayerMode::File => {
+                let config_file_path =
+                    std::env::var("TOKEN_PRICES_JSON").unwrap_or_else(|_| "token_prices.json".to_owned());
+
+                let config_str = match tokio::fs::read_to_string(config_file_path).await {
+                    Err(err) => {
+                        error!("Error while fetching token prices {}", err);
+                        return false;
+                    }
+                    Ok(value) => value,
+                };
+
+                let config: HashMap<String, String> = match serde_json::from_str(&config_str)
+                {
+                    Err(err) => {
+                        error!("Error while parsing token prices json configuration: {}", err);
+                        return false;
+                    }
+                    Ok(config) => config,
+                };
+
+                self.token_price_map = config;
+                return true;
+            }
+            _ => {
+                return true;
+            }
+        }
+    }
+
+    // A batch can be send either if
+    // - Mode is AlwaysRelay
+    // - Mode is either API or File and the batch has a profitable cost
+    // - Mode is either API or File and the batch has been waiting to be sent more than GRAVITY_BATCH_SENDING_SECS secs
+    pub async fn can_send_batch(
+        &mut self,
+        estimated_cost: &GasCost,
+        batch_fee: &Erc20Token,
+        contract_address: &EthAddress,
+    ) -> bool {
+        if self.mode == RelayerMode::AlwaysRelay {
+            return true;
+        }
+
+        match self.next_batch_send_time.get(contract_address) {
+            Some(time) => {
+                if *time < Instant::now() {
+                    return true;
+                }
+            }
+            None => self.update_next_batch_send_time(*contract_address),
+        }
+
+        let token_price = match self.get_token_price(&batch_fee.token_contract_address).await {
+            Ok(token_price) => token_price,
+            Err(_) => return false,
+        };
+
+        let estimated_fee = estimated_cost.get_total();
+        let batch_value = batch_fee.amount.clone() * token_price;
+
+        info!("estimate cost is {}, batch value is {}", estimated_fee, batch_value);
+        batch_value >= estimated_fee
+    }
+
+    pub fn update_next_batch_send_time(
+        &mut self,
+        contract_address: EthAddress,
+    ) {
+        if self.mode == RelayerMode::AlwaysRelay {
+            return;
+        }
+
+        let timeout_duration = std::env::var("GRAVITY_BATCH_SENDING_SECS")
+            .map(|value| Duration::from_secs(value.parse().unwrap()))
+            .unwrap_or_else(|_| Duration::from_secs(3600));
```

Comment:
remove `unwrap()`, return `Error`

Reply:
I think it is okay to have default value?